File: cmManagementSystem/src/main/java/com/example/cmManagementSystem/service/ReportService.java
package com.example.cmManagementSystem.service;

import com.example.cmManagementSystem.dto.ReportDto;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Map;

/**
 * Raporlama işlemlerini yöneten servis
 */
public interface ReportService {
    
    /**
     * Tüm raporları getirir
     */
    List<ReportDto> findAll();
    
    /**
     * ID'ye göre rapor getirir
     */
    ReportDto findById(Long id);
    
    /**
     * Kullanıcıya göre raporları getirir
     */
    List<ReportDto> findByUser(Long userId);
    
    /**
     * Kulübe göre raporları getirir
     */
    List<ReportDto> findByClub(Long clubId);
    
    /**
     * Etkinliğe göre raporları getirir
     */
    List<ReportDto> findByEvent(Long eventId);
    
    /**
     * Yeni rapor oluşturur
     */
    ReportDto createReport(ReportDto reportDto);
    
    /**
     * Kulüp için rapor oluşturur
     */
    ReportDto generateClubReport(Long clubId, String startDate, String endDate, String type);
    
    /**
     * Etkinlik için rapor oluşturur
     */
    ReportDto generateEventReport(Long eventId, String type);
    
    /**
     * Raporu günceller
     */
    ReportDto updateReport(Long id, ReportDto reportDto);
    
    /**
     * Raporu siler
     */
    void deleteReport(Long id);
    
    /**
     * Dashboard istatistiklerini getirir
     */
    Map<String, Object> getDashboardStats();
    
    /**
     * Raporu PDF olarak dışa aktarır
     */
    ResponseEntity<?> exportReportAsPdf(Long id);
    
    /**
     * Raporu Excel olarak dışa aktarır
     */
    ResponseEntity<?> exportReportAsExcel(Long id);
    
    /**
     * Rapor belirli bir kullanıcı tarafından oluşturulmuş mu kontrol eder
     */
    boolean isReportCreatedByUser(Long userId, Long reportId);
    
    /**
     * Kullanıcının rapora erişimi var mı kontrol eder
     */
    boolean isReportAccessibleByUser(Long userId, Long reportId);
} 

File: src/services/reportService.js
import { api } from './api';

/**
 * Rapor işlemleri ile ilgili servis
 */

// Etkinlik raporu oluştur/getir
export const generateEventReport = async (eventId) => {
  try {
    const response = await api.generateEventReport(eventId);
    return response.data;
  } catch (error) {
    console.error(`Etkinlik raporu oluşturulurken hata oluştu (Etkinlik ID: ${eventId}):`, error);
    throw new Error(error.response?.data?.message || 'Etkinlik raporu oluşturulamadı');
  }
};

// Kulüp raporu oluştur/getir
export const generateClubReport = async (clubId) => {
  try {
    const response = await api.generateClubReport(clubId);
    return response.data;
  } catch (error) {
    console.error(`Kulüp raporu oluşturulurken hata oluştu (Kulüp ID: ${clubId}):`, error);
    throw new Error(error.response?.data?.message || 'Kulüp raporu oluşturulamadı');
  }
};

// Kullanıcı raporu oluştur/getir
export const generateUserReport = async (userId) => {
  try {
    const response = await api.generateUserReport(userId);
    return response.data;
  } catch (error) {
    console.error(`Kullanıcı raporu oluşturulurken hata oluştu (Kullanıcı ID: ${userId}):`, error);
    throw new Error(error.response?.data?.message || 'Kullanıcı raporu oluşturulamadı');
  }
};

// Tüm raporları getir
export const getAllReports = async () => {
  try {
    const response = await api.getAllReports();
    return response.data;
  } catch (error) {
    console.error('Raporlar yüklenirken hata oluştu:', error);
    return [];
  }
};

// ID'ye göre rapor detaylarını getir
export const getReportById = async (reportId) => {
  try {
    const response = await api.getReportById(reportId);
    return response.data;
  } catch (error) {
    console.error(`Rapor detayları yüklenirken hata oluştu (ID: ${reportId}):`, error);
    return null;
  }
};

// Özel rapor oluştur
export const createCustomReport = async (reportData) => {
  try {
    const response = await api.createCustomReport(reportData);
    return response.data;
  } catch (error) {
    console.error('Özel rapor oluşturulurken hata oluştu:', error);
    throw new Error(error.response?.data?.message || 'Özel rapor oluşturulamadı');
  }
}; 